suite-native/device/src/middlewares/buttonRequestMiddleware.ts

```diff
@@ -17,8 +18,10 @@ export const prepareButtonRequestMiddleware = createMiddlewareWithExtraDeps(
             );
         }
 
-        if (action.type === UI.REQUEST_BUTTON) {
+        // Pin is not handled by buttonRequest but redux state within @suite-native/device-authentication
+        if (action.type === UI.REQUEST_BUTTON && !isPinButtonRequestCode(action)) {
```

Comment:
Hmm. I was thinking that this button request will be unused in suite-native and if we switch to this state management, it would just be a clutter. Elsewhere, it could interfere with some other logic. No hard opinion on this, but I just think it would be redundant to keep it.

Reply:
The status quo is that any REQUEST_BUTTON is added as buttonRequest to the device. It was like that before we started using buttonRequests array for PIN handling. I would prefer to not exclude them here. It's nice at least for debugging.

Reply:
https://github.com/trezor/trezor-suite/pull/12889/commits/d12344e0a9cb6d9a3dcdb4ee3eea5c6674ceaf09